add rwn optimizations

jnunyez opened the pull request:
Generic rwn-optimizations: Moved from vse-carslab-managed to vse-catalog.

novacain1 commented on components/configs/rwn-optimizations/base/3-node/config-ingress.yaml:
```diff
@@ -0,0 +1,12 @@
+---
+apiVersion: operator.openshift.io/v1
+kind: IngressController
```
One could also use Router sharding here to affect the behavior of where the ingress sits, if it is not wanted on control plane hosts / supervisors and instead on the RWN: https://docs.openshift.com/container-platform/4.12/networking/ingress-operator.html#nw-ingress-sharding_configuring-ingress

Safe to say this is outside the scope for right now, but sharing nonetheless.

jnunyez replied:
Agree in that  out of scope. I can see use case ingress controller in edge worker node in real deployment assuming full cluster when RAN outside of OCP domain.

novacain1 commented on components/configs/rwn-optimizations/base/3-node/master-schedulable.yaml:
```diff
@@ -0,0 +1,7 @@
+metadata:
+  name: cluster
+spec:
+  mastersSchedulable: true
```
I believe this is true already if one deploys a Compact Cluster -- 3 node?
I think this might be a leftover from when we had to deploy a Full cluster -- 3 control plane, 2 workers and one wanted to specifically make the control plane scheduleable.

jnunyez replied:
Good point. Taking that one out.

novacain1 commented on components/configs/rwn-optimizations/base/3-node/worker-fix-ipi-rwn.yaml:
```diff
@@ -0,0 +1,46 @@
+          filesystem: root
+          mode: 420
+          path: /etc/kubernetes/manifests/coredns.yaml
+    systemd:
+      units:    # yamllint disable rule:line-length
+        - contents: |
+            [Unit]
+            Description=Writes IP address configuration so that kubelet and crio services select a valid node IP
+            Wants=network-online.target
+            After=network-online.target ignition-firstboot-complete.service
+            Before=kubelet.service crio.service
+            [Service]
+            Type=oneshot
+            ExecStart=/bin/bash -c "exit 0 "
+            [Install]
+            WantedBy=multi-user.target
+          enabled: true   # yamllint disable rule:line-length
+          name: nodeip-configuration.service
```
I really don't think this is needed any longer, we should really test this.

jnunyez replied:
That is a doubt I had. I remember you bringing this issue when doing an upgrade in OCP 4.10. I can take a look and test with  the SNOs I have (OCP 4.11 and OCP 4.12).

jnunyez replied:
@novacain1 Looking good in SNO OCP 4.11 and 4.12. No need of that file. This is now shipped as base template in mco:
```
[Unit]
Description=Writes IP address configuration so that kubelet and crio services select a valid node IP
Wants=NetworkManager-wait-online.service crio-wipe.service
After=NetworkManager-wait-online.service ignition-firstboot-complete.service crio-wipe.service
Before=kubelet.service crio.service ovs-configuration.service

[Service]
# Need oneshot to delay kubelet
Type=oneshot
# Would prefer to do Restart=on-failure instead of this bash retry loop, but
# the version of systemd we have right now doesn't support it. It should be
# available in systemd v244 and higher.
ExecStart=/bin/bash -c " \
  until \
  /usr/bin/podman run --rm \
  --authfile /var/lib/kubelet/config.json \
  --net=host \
  --security-opt label=disable \
  --volume /etc/systemd/system:/etc/systemd/system \
  --volume /run/nodeip-configuration:/run/nodeip-configuration \
  quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:6b61429030b790e6ec6e9fcb52b2a17c5b794815d6da9806bc563bc45e84aa67 \
  node-ip \
  set \
  --retry-on-failure \
  ${NODEIP_HINT:-${KUBELET_NODEIP_HINT:-}}; \
  do \
  sleep 5; \
  done"
ExecStart=/bin/systemctl daemon-reload
ExecStartPre=/bin/mkdir -p /run/nodeip-configuration

EnvironmentFile=-/etc/default/nodeip-configuration

[Install]
RequiredBy=kubelet.service
```

novacain1 commented on components/configs/rwn-optimizations/base/3-node/worker-fix-ipi-rwn.yaml:
```diff
@@ -0,0 +1,46 @@
+          filesystem: root
+          mode: 420
+          path: /etc/kubernetes/manifests/mdns-publisher.yaml
+        - contents:
+            source: data:,
+            verification: {}
+          filesystem: root
+          mode: 420
+          path: /etc/kubernetes/manifests/coredns.yaml
```
This too I think is no longer necessary post 4.8.  I believe originally coredns generated mDNS traffic in the cluster, which wasn't something we wanted.

jnunyez replied:
@novacain1 I believe this should be checked with a compact cluster.    I would propose to keep the keepalived.yaml since it is still needed in OCP 4.10, and add note and comment the others that are not needed.

natifridman reviewed (commented):
Since I see `machineconfig` here, do we care about the ordering?

jnunyez commented:
> Since I see `machineconfig` here, do we care about the ordering?

I think not in this specific case. We need to try.

jnunyez reviewed (commented):
This RWN resulting config needs to be tested in compact cluster (OCP 4.10 minimum).

natifridman reviewed (commented):
Looks good. Could you please fix the yamllint warnings?

jnunyez commented:
> 

> Looks good. Could you please fix the yamllint warnings?

Done.

jnunyez commented:
Rebased with main and squashed my changed onto one commit.  I'm not sure we did try all the RWN changes in compact cluster. We can also try afterwards to unblock vse-carslab-managed repo PR.

novacain1 reviewed (approved):
Looks good.
